video/out/wayland_common.c

```diff
@@ -1416,28 +1440,26 @@ static void tranche_formats(void *data,
 {
     struct vo_wayland_state *wl = data;
 
-    // Only grab formats from the first tranche and ignore the rest.
-    if (!wl->add_tranche)
-        return;
-
     // Should never happen.
     if (!wl->compositor_format_map) {
         MP_WARN(wl, "Compositor did not send a format and modifier table!\n");
         return;
     }
 
-    const compositor_format *formats = wl->compositor_format_map;
-    MP_RESIZE_ARRAY(wl, wl->compositor_formats, indices->size);
-    wl->num_compositor_formats = 0;
+    struct vo_wayland_tranche *tranche = wl->current_tranche;
+    if (!tranche)
+        return;
+
+    const struct compositor_format *formats = wl->compositor_format_map;
     uint16_t *index;
     wl_array_for_each(index, indices) {
-        MP_TARRAY_APPEND(wl, wl->compositor_formats, wl->num_compositor_formats,
+        MP_TARRAY_APPEND(tranche, tranche->compositor_formats, tranche->num_compositor_formats,
                          (struct drm_format) {
                             formats[*index].format,
                             formats[*index].modifier,
                         });
-        MP_DBG(wl, "Compositor supports drm format: '%s(%016" PRIx64 ")'\n",
-               mp_tag_str(formats[*index].format), formats[*index].modifier);
+        MP_DBG(wl, "Compositor supports drm format: '%s(%016" PRIx64 ")' on device '%lu'\n",
```

Comment:
Could we extend this log to say also which tranche the format is coming from? Would be good for debugging multiple tranche cases.

P.S. In the future we could add an option to allow select both device and tranche so that we limit the formats used to given group. I think doing it automatically may be too hard in multi-gpu / multi-tranche scenario where formats doesn't line up as we want the to. I know it would be low-level option, but whole premise of this vo is to do direct rendering, so we should make it possible :)

Reply:
Could just increment an index to identify tranches I guess?

Reply:
Should work, as long as we know in which order they came from compositor.

Reply:
I just pushed an index, but maybe just printing some header is nicer.